### DISCUSSION TITLE
Why removed custom routes？

### DISCUSSION THREAD

**Discussion by Lisheri** [nuxt/nuxt#16862] [Questions] [answered]
### Describe the feature

The part about custom routes in nuxt2 is very important. Fixing to a configuration that cannot be customized is not what developers want. Losing this function may make most developers lose interest in nuxt3. After all, compared to nuxt, next based on the react technology stack has more complete functions and supports flexible configuration.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by pi0** [comment, marked as answer]
Please check the migration guide: https://v3.nuxtjs.org/migration/configurationCustom routes is possible only differently using `pages:extend` hook:

Example: https://stackblitz.com/edit/github-eevymn?file=routes%2Ftest.vue,routes%2Findex.vue,app.vue,nuxt.config.ts


**New:** Using new `router.addRoute` for runtime custom routes:

Docs: https://v3.nuxtjs.org/api/composables/use-router

Example: https://stackblitz.com/edit/github-eevymn-tyt5rh?file=routes%2Ftest.vue,routes%2Findex.vue,nuxt.config.ts,plugins%2Froutes.ts

**Upcoming:** Using new `app/router.config.ts` to declare custom routes:

PR: https://github.com/nuxt/framework/pull/7129

**Reply by Lisheri** [reply]
I know this hook, but I feel a bit hacky in this way, and can't hot update.
In addition, I see that there is a Note in the document, which means that the routing configuration of nuxt cannot be overwritten.
![image](https://user-images.githubusercontent.com/49238808/188301467-1d896c46-a859-4941-b636-87c15e10e088.png)

**Reply by Lisheri** [reply]
When will the routing configuration be released?

**Reply by pi0** [reply]
In next RC version :)

**Reply by Lisheri** [reply]
> In next RC version :)

nice~

**Reply by tasiotas** [reply]
@pi0 is it in RC10?

**Reply by danielroe** [reply]
Yes.
